library/alloc/src/collections/vec_deque.rs

```diff
@@ -47,10 +47,17 @@ const MAXIMUM_ZST_CAPACITY: usize = 1 << (64 - 1); // Largest possible power of
 /// push onto the back in this manner, and iterating over `VecDeque` goes front
 /// to back.
 ///
+/// Since `VecDeque` is a ring buffer, its elements are not necessarily contiguous
+/// in memory. If you want to access the elements as a single slice, such as for
+/// efficient sorting, you can use [`make_contiguous`]. It rotates the `VecDeque`
```

Comment:
Hm -- I would expect that we could implement sorting almost as efficiently on two slices as one? We still have Index/IndexMut impls, so anything a sort would need to do seems like it should be essentially just as fast.

Reply:
A sort impl for `VecDeque` was submitted in https://github.com/rust-lang/rust/pull/69400#issuecomment-597644295, but was then abandoned in favor of `make_contiguous`. I agree with you that it's probably possible to implement an efficient sort directly on a `VecDeque`, though it's certainly a bit more complicated (and would also add many extra methods). For example, a straightforward heapsort wouldn't work unless you do some tricky wrapping index calculation for child/parent indexing. Or you could go for two heapsorts + a merge (which I think the original PR did), but that is also a bit slower.

Ultimately, I think `make_contiguous` is useful regardless of whether we get sorting. And this doc text can always be updated to reflect added sorting methods if those get added later.

Reply:
:+1: seems reasonable.